src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,337 @@
+            Components::GrainEstimatorImpl<GrainType, AnalyticalRollingWindowEstimator<InputT, MaxNumTrainingItemsV>>,
+            Components::FilterDecoratorEstimator<std::tuple<GrainType const &, std::vector<double>>, 1>
+        >;
+    
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    // MSVC has problems when the definition for the constructor is separated from its declaration.
+    GrainedAnalyticalRollingWindowEstimator(AnnotationMapsPtr pAllColumnAnnotations, std::uint32_t maxWindowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowSize = 1);
+    ~GrainedAnalyticalRollingWindowEstimator(void) override = default;
+};
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  AnalyticalRollingWindowTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename InputT, size_t MaxNumTrainingItemsV>
+AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::AnalyticalRollingWindowTransformer(std::uint32_t maxWindowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowSize) :
+    BaseType(
+        std::move(maxWindowSize), 
+        [] (typename Components::CircularBuffer<InputT>::iterator begin, typename Components::CircularBuffer<InputT>::iterator end) {
+            return Calculators::MeanCalculator<InputT>::execute(begin, end);
+        },
+        std::move(horizon),
+        std::move(minWindowSize)),
+        _windowCalculation(std::move(windowCalculation))
+    {
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::AnalyticalRollingWindowTransformer(Archive &ar) :
+    AnalyticalRollingWindowTransformer(
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Because we know that maxWindowSize and minWindowSize come in originally as uint32, this cast is always safe

```

Comment:
>// Because we know that maxWindowSize and minWindowSize come in originally as uint32, this cast is always safe [](start = 12, length = 110)

Does this comment still apply?

Reply:
Nope, I will remove it.